File: Utilities/Matlab/functions/friction_factor_calc.m
% McDermott
% 5-14-2009
% friction_factor_calc.m

function [f_fds,Re_H]=friction_factor_calc(dpdx,H,filename,varargin)

M = csvread(filename,2,0);

ubar = M(:,2);
if nargin==3
    mu = max(M(:,4));
elseif nargin==4
    mu = varargin{1};
end

rho = max(M(:,5));

U = ubar(length(ubar));  % steady state mean velocity (planar averaged)
Re_H = H*U*rho/mu;              % Reynolds number based on H
f_fds = 2*(-dpdx)*H/(rho*U^2);  % f from FDS

File: Utilities/Matlab/pythonscripts/friction_factor_calc.py
import numpy as np
# McDermott
# 5-14-2009
# friction_factor_calc.m


def friction_factor_calc(dpdx=None,H=None,filename=None,mu_in=None):

    M=np.nan_to_num(np.genfromtxt(filename, delimiter = ',')[2:,0:], copy = False)
# temp/friction_factor_calc.m:7
    ubar=M[:,1]
# temp/friction_factor_calc.m:9
    if mu_in == None:
        mu=max(M[:,3])
# temp/friction_factor_calc.m:11
    else:
        mu=mu_in
# temp/friction_factor_calc.m:13
    
    rho=max(M[:,4])
# temp/friction_factor_calc.m:16
    U=ubar[ubar.size-1]
# temp/friction_factor_calc.m:18
    
    Re_H=np.dot(np.dot(H,U),rho) / mu
# temp/friction_factor_calc.m:19
    
    f_fds=np.dot(np.dot(2,(- dpdx)),H) / (np.dot(rho,U ** 2))
# temp/friction_factor_calc.m:20
    
    return f_fds,Re_H